src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -62,6 +62,11 @@ private static bool ProxySupportsConnectionAuth(HttpResponseMessage response)
                         while (true)
                         {
                             string challengeResponse = authContext.GetOutgoingBlob(challengeData);
+                            if (challengeResponse == null)
+                            {
+                                // Server sent something invalid, so stop processing and return current response.
```

Comment:
The server's response wasn't invalid. It indicated denial even after login. What was invalid was the following request with no data blob, which this new condition handles.